test/check-application

```diff
@@ -2065,6 +2069,110 @@ class TestFiles(testlib.MachineCase):
+
+        # Paste as admin into foouser directory
+        m.execute("runuser -u admin touch /home/admin/adminfile")
+        b.wait_in_text("[data-item='adminfile'] .item-owner", "admin")
+        b.click("[data-item='adminfile']")
+        b.click("#dropdown-menu")
+        b.click("#copy-item")
+        b.go("files#/?path=/home/foouser")
+        self.assert_last_breadcrumb("foouser")
+        b.click("#dropdown-menu")
+        b.click("#paste-item")
+        b.wait_visible("#paste-owner-modal")
+        b.select_from_dropdown("#paste-owner-select", "admin (original owner)")
+        b.click("#paste-owner-modal button.pf-m-primary")
+        b.wait_in_text("[data-item='adminfile'] .item-owner", "admin")
+
+        # Copying a directory should change the owner recursively
+        b.go("files#/?path=/home/admin")
+        self.assert_last_breadcrumb("admin")
+        b.click("[data-item='kbdCopy']")
+        b.click("#dropdown-menu")
+        b.click("#copy-item")
+        b.go("files#/?path=/home/foouser")
+        self.assert_last_breadcrumb("foouser")
+        b.click("#dropdown-menu")
+        b.click("#paste-item")
+        b.wait_visible("#paste-owner-modal")
+        b.select_from_dropdown("#paste-owner-select", "root")
+        b.click("#paste-owner-modal button.pf-m-primary")
+        b.wait_in_text("[data-item='kbdCopy'] .item-owner", "root")
+        b.go("files#/?path=/home/foouser/kbdCopy")
+        self.assert_last_breadcrumb("kbdCopy")
+        b.wait_in_text("[data-item='loaded'] .item-owner", "root")
+        b.wait_in_text("[data-item='newfile'] .item-owner", "root")
+
+        # Copy file with different user, group
+        b.go("files#/?path=/home/admin")
+        self.assert_last_breadcrumb("admin")
+        m.execute("""
+            touch /home/admin/adminfoofile
+            chown admin:foouser /home/admin/adminfoofile
+        """)
+        b.click("[data-item='adminfoofile']")
+        b.click("#dropdown-menu")
+        b.click("#copy-item")
+        b.go("files#/?path=/home/foouser")
+        self.assert_last_breadcrumb("foouser")
+        b.click("#dropdown-menu")
+        b.click("#paste-item")
+        b.wait_visible("#paste-owner-modal")
+        b.select_from_dropdown("#paste-owner-select", "admin:foouser (original owner)")
+        b.click("#paste-owner-modal button.pf-m-primary")
+        b.wait_in_text("[data-item='adminfoofile'] .item-owner", "admin:foouser")
+
+        b.go("files#/?path=/home/admin")
+        self.assert_last_breadcrumb("admin")
+        b.click("[data-item='newdir']")
+        b.click("#dropdown-menu")
+        b.click("#copy-item")
+        b.go("files#/?path=/home/foouser")
+        self.assert_last_breadcrumb("foouser")
+        b.click("#dropdown-menu")
+        b.click("#paste-item")
+        b.wait_visible("#paste-owner-modal")
+        b.click("button.pf-m-link")
+        b.wait_not_present("#paste-owner-modal")
+        b.wait_not_present("[data-item='newdir']")
```

Comment:
Please also cover copying a directory tree (to make sure the recursive cp/chown works), and a directory with mixed owners and permissions. This is related to the discussion above, but we should at least pin down the behaviour (i.e. right now let them all get chowned).